modules/zone_association/main.tf

```
@@ -0,0 +1,6 @@
+resource "aws_route53_zone_association" "this" {
+  vpc_id  = var.vpc_id.vpc_id
```

In conjunction with the change to the variable object:

```suggestion
  vpc_id  = var.vpc_id
```